visualizer/visualizer.py

```diff
@@ -324,30 +325,92 @@ def add_html_framework(self, maindata):
+        csv_list = []
+        title_row = []
+        title_row = data[data.keys()[0]].keys()
+        title_row.insert(0,'')
+        csv_list.append(title_row)
+        for key,value in data.items():
+            row = []
+            row.append(key)
+            for i in range(len(value)):
+                row.append(value[title_row[i+1]])
+            csv_list.append(row)
+        file_name = "ceph_%s_detail_data.csv"%field_type
+        csv_path = os.path.join(self.db_path,session_name,"include/csv/",file_name)
+        csvfile = file(csv_path, 'wb')
+        csv_writer = csv.writer(csvfile)
+        for row in csv_list:
+            csv_writer.writerow(row)
+
     def generate_node_view(self, node_type):
         output = []
         if len(self.result[node_type].keys()) == 0:
             return output
         output.append("<div id='%s'>" % node_type)
-        for field_type, field_data in self.result[node_type].items():
-            data = OrderedDict()
-            chart_data = OrderedDict()
-            for node, node_data in field_data.items():
-                if node not in data:
-                    data[node] = OrderedDict()
-                for key, value in node_data.items():
-                    if not isinstance(value, list):
-                        data[node][key] = value
-                    else:
-                        data[node][key] = "%.3f" % numpy.mean(value)
-                        if key not in chart_data:
-                            chart_data[key] = OrderedDict()
-                        chart_data[key][node] = value
-            output.extend( self.generate_table_from_json(data,'cetune_table', field_type) )
-            output.extend( self.generate_line_chart(chart_data, node_type, field_type ) )
+        if node_type == "ceph":
+            for field_type, field_data in self.result[node_type].items():
+                summary_data = OrderedDict()
+                detail_data = OrderedDict()
+                chart_data = OrderedDict()
+                for node, node_data in field_data["summary"].items():
+                    if node not in summary_data:
+                        summary_data[node] = OrderedDict()
+                    for key, value in node_data.items():
+                        if not isinstance(value, list):
+                            summary_data[node][key] = value
+                        else:
+                            summary_data[node][key] = "%.3f" % numpy.mean(value)
+                            if key not in chart_data:
+                                chart_data[key] = OrderedDict()
+                            chart_data[key][node] = value
+                if len(field_data["detail"]) != 0:
+                    for node, node_data in field_data["detail"].items():
+                        if node not in detail_data:
+                            detail_data[node] = OrderedDict()
+                        for key, value in node_data.items():
```

Comment:
this part of codes is duplicated with summary, can you merge it? Or make it as one function？